src/Framework/ClassResolver/DependencyResolver/DependencyException.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gacela\Framework\ClassResolver\DependencyResolver;
+
+use RuntimeException;
+
+final class DependencyException extends RuntimeException
+{
+    public static function noParameterTypeFor(string $parameter): self
+    {
+        return new self("No parameter type for '{$parameter}'");
+    }
+
+    public static function unableToResolve(string $parameter, string $className): self
+    {
+        return new self("Unable to resolve [{$parameter}] from {$className}");
+    }
```

Comment:
What about splitting this class into two?

1. A new class that extends from `InvalidArgumentException` class for `noParameterTypeFor()` and `unableToResolve()` methods
2. Another that extends from `RuntimeException` for the `mapNotFoundForClassName()` method

Reply:
Nice idea. Feel free to do it yourself in this branch.